src/JWT/JwtDecoder.cs

```diff
@@ -122,5 +110,28 @@ public void Validate(string[] parts, byte[] key)
 
             _jwtValidator.Validate(payloadJson, decodedCrypto, decodedSignature);
         }
+
+        /// <summary>
+        /// Get the Jwt token as string and return a parts array of <see cref="string[]"/> object.
+        /// </summary>
+        /// <param name="token">The JWT.</param>
+        /// <returns>The converted parts array of <see cref="string[]"/></returns>
+        /// <exception cref="ArgumentException">Thrown if the given token have the wrong fromat.</exception>
+        private string[] splitToParts(string token)

```

Comment:
Please use UpperCase for the name and make it static.

Reply:
And let's return `IDictionary<JwtParts, stirng>` rather than `string[]` to avoid casting enum to int over and over again.

Reply:
done